render/render.js

```diff
@@ -312,22 +315,22 @@ module.exports = function($window) {
 				else if (o == null) oldEnd--
 				else if (v == null) end--
 				else if (o.key === v.key) {
-					updateNode(parent, o, v, hooks, getNextSibling(old, oldEnd + 1, nextSibling), ns)
-					if (o.dom != null) nextSibling = o.dom
+					updateNode(parent, o, v, hooks, nextSibling, ns)
+					if (v.dom != null) nextSibling = v.dom
 					oldEnd--, end--
 				} else {
 					if (!map) map = getKeyMap(old, oldEnd)
 					if (v != null) {
 						var oldIndex = map[v.key]
 						if (oldIndex != null) {
 							o = old[oldIndex]
-							updateNode(parent, o, v, hooks, getNextSibling(old, oldEnd + 1, nextSibling), ns)
-							insertNode(parent, toFragment(v), nextSibling)
+							insertNode(parent, toFragment(o), nextSibling)
+							updateNode(parent, o, v, hooks, nextSibling, ns)
```

Comment:
Just below, replacing `o.skip = true` with `old[oldIndex] = null` works as well, and it would save us a vnode slot.

Is there a reason not to mutate the old children list?

Reply:
> save us a vnode slot

Not sure I follow...

Reply:
Sorry, that wasn't clear. If we do that, we can remove the `.skip` field on vnodes, improving their memory footprint.